UI/MapWnd.cpp

```diff
@@ -283,6 +284,198 @@ namespace {
+            const GG::Y offset{32};
+            const GG::Clr& BG_CLR = ClientUI::WndColor();
+            const GG::Clr& BORDER_CLR = ClientUI::WndOuterBorderColor();
+            const GG::Pt& UL = GG::Pt(UpperLeft().x, UpperLeft().y + offset);
+            const GG::Pt& LR = LowerRight();
+
+            // main background
+            GG::FlatRectangle(UL, LR, BG_CLR, BORDER_CLR);
+
+            // summary text background
+            GG::FlatRectangle(UL, GG::Pt(LR.x, row_height + offset), BORDER_CLR, BORDER_CLR);
+
+            // Seperation line between armed/unarmed and utility ships
+            GG::Y line_ht(UL.y + (row_height * 2) + (row_height * 5 / 4));
+            GG::Pt line_ul(UL.x + (m_margin * 2), line_ht);
+            GG::Pt line_lr(LR.x - (m_margin * 2), line_ht);
+            GG::Line(line_ul, line_lr, BORDER_CLR);
+
+            // inset border for parts/slots
+            GG::Pt part_ul(UL.x + m_margin, LR.y - ((m_margin + row_height) * 2));
+            GG::Pt part_lr(LR.x - m_margin, LR.y - m_margin);
+            GG::BeveledRoundedRectangle(part_ul, part_lr, BG_CLR, BORDER_CLR, false);
+        }
+
+        void DoLayout() {
+            const GG::Y row_height{ClientUI::Pts()};
+            const GG::Y offset{32};
+            const GG::X label_width{m_col_widths.at(0) - (m_margin * 2)};
+            const GG::X value_width{m_col_widths.at(1) - (m_margin * 3)};
+
+            GG::Pt label_ul{GG::X{m_margin}, offset + m_margin};
+            GG::Pt label_lr{label_ul.x + label_width, offset + row_height};
+            GG::Pt value_ul{label_lr.x + m_margin, label_ul.y};
+            GG::Pt value_lr{value_ul.x + value_width, label_lr.y};
+
+            const GG::Pt next_row{GG::X0, row_height + (m_margin * 2)};
+            const GG::Pt space_row{next_row * 5 / 4};
+
+            LayoutRow("MAP_FLEET_SHIP_COUNT",       label_ul, label_lr, value_ul, value_lr, space_row);
+            LayoutRow("MAP_FLEET_ARMED_COUNT",      label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_UNARMED_COUNT",    label_ul, label_lr, value_ul, value_lr, space_row);
+            LayoutRow("MAP_FLEET_CARRIER_COUNT",    label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_TROOP_COUNT",      label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_COLONY_COUNT",     label_ul, label_lr, value_ul, value_lr, space_row);
+            LayoutRow("MAP_FLEET_SLOT_COUNT",       label_ul, label_lr, value_ul, value_lr, next_row);
+            LayoutRow("MAP_FLEET_PART_COUNT",       label_ul, label_lr, value_ul, value_lr, GG::Pt(GG::X0, GG::Y0));
+
+            Resize(GG::Pt(value_lr.x + (m_margin * 3), value_lr.y + (m_margin * 3)));
+        }
+
+        /** Constructs and attaches new label and value controls for the given label. */
+        void NewLabelValue(const std::string& label, bool title = false) {
```

Comment:
label and m_labels are a bit confusing. rename label to something more explicit, like label_text or label_name or whatever is appropriate